packages/@react-spectrum/card/src/GridLayout.tsx

```diff
@@ -147,13 +147,31 @@ export class GridLayout<T> extends BaseLayout<T> {
       // The approach from v2 uses indexes where other v3 layouts iterate through every node/root node. This feels more efficient
       let firstVisibleItem = this.getIndexAtPoint(rect.x, rect.y);
       let lastVisibleItem = this.getIndexAtPoint(rect.maxX, rect.maxY);
+      // Check to see if an item with a persisted key exists after the visible items
+      // This is for keeping focus on a item that is scrolled out of view
+      for (let h = 0; h < firstVisibleItem; h++) {
+        let keyFromIndex = this.collection.rows[h].key;
+        let layoutInfo = this.layoutInfos.get(keyFromIndex);
+        if (layoutInfo && this.virtualizer.isPersistedKey({layoutInfo: layoutInfo})) {
+          res.push(layoutInfo);
+        }
+      }
       for (let index = firstVisibleItem; index <= lastVisibleItem; index++) {
         let keyFromIndex = this.collection.rows[index].key;
         let layoutInfo = this.layoutInfos.get(keyFromIndex);
         if (layoutInfo && this.isVisible(layoutInfo, rect)) {
           res.push(layoutInfo);
         }
       }
+      // Check to see if an item with a persisted key exists after the visible items
+      // This is for keeping focus on a item that is scrolled out of view
+      for (let j = lastVisibleItem; j < this.collection.size; j++) {
+        let keyFromIndex = this.collection.rows[j].key;
+        let layoutInfo = this.layoutInfos.get(keyFromIndex);
+        if (layoutInfo && this.virtualizer.isPersistedKey({layoutInfo: layoutInfo})) {
+          res.push(layoutInfo);
+        }
+      }
```

Comment:
same overlap in this for loop and the loop before it when checking `lastVisibleItem`

Reply:
I was concerned about this, but when I do something like `for (let j = lastVisibleItem + 1; j < this.collection.size; j++) {` and you've focused the last item in the grid, the last time gets removed from visible layouts when it shouldn't. I want to look more closely at this.